Update the telco feature config (#723)
suse-edge/suse-edge.github.io: asciidoc/product/atip-features.adoc

+
This behavior prevents any disruption caused by disk I/O to any time sensitive application running on the isolated cores, but might require attention and careful design for storage focused workloads.

* Tune the ticks (kernel's periodic timer interrupts):

> ** `skew_tick=1`: ticks can sometimes happen simultaenously. Instead of all CPUs receiving their timer tick at the exact same moment, `skew_tick=1` makes them occur at slightly offset times. This smoothes out system jitter and lead to more predictable and lower interrupt response times, which is critical for latency-sensitive apps.

Reviewer: ```suggestion
** `skew_tick=1`: ticks can sometimes happen simultaenously. Instead of all CPUs receiving their timer tick at the exact same moment, `skew_tick=1` makes them occur at slightly offset times. This helps reduce system jitter, resulting in more consistent and lower interrupt response times (an essential requirement for latency-sensitive applications).
```

** `nohz=on`: stops the periodic timer tick on idle CPUs.

> ** `nohz_full=<cpu-cores>`: Stops the period timer tick on specified CPUs that are dedicated for applications.

Reviewer: ```suggestion
** `nohz_full=<cpu-cores>`: Stops the period timer tick on specified CPUs that are dedicated for real-time applications.
```

* Disable Machine Check Exception (MCE) handling by specifying `mce=off`. MCEs are hardware errors detected by the processor and disabling them can avoid noisy logs.

* Add `nowatchdog` to disable the soft-lockup watchdog which is implemented as a timer running in the timer hard-interrupt context. When it expires (i.e. a soft lockup is detected), it will print a warning (in the hard interrupt context), running any latency targets. Even if it never expires, it goes onto the timer list, slightly increasing the overhead of every timer interrupt. This option also disables the NMI watchdog, so NMIs cannot interfere.

* `nmi_watchdog=0` disables the NMI (Non-Maskable Interrupt) watchdog. This can be omitted when `nowatchdog` is used.

* RCU (Read-Copy-Update) is a kernel mechanism that enables concurrent, lock-free access for many readers to shared data. An RCU callback, a function triggered after a 'grace period', ensures all previous readers have finished so old data can be safely reclaimed. We fine-tune RCU, particularly for sensitive workloads, to offload these callbacks from dedicated (pinned) CPUs, preventing kernel operations from interfering with critical, time-sensitive tasks.

> ** Specify the pinned CPUs in `rcu_nocbs` so that RCU callbacks do not run on them. This helps reducing jitter and latency for the workloads.

Reviewer: ```suggestion
** Specify the pinned CPUs in `rcu_nocbs` so that RCU callbacks do not run on them. This helps reducing jitter and latency for the real-time workloads.
```

[,shell]
----
$ cat /proc/cmdline
BOOT_IMAGE=/boot/vmlinuz-6.4.0-9-rt root=UUID=575291cf-74e8-42cf-8f2c-408a20dc00b8 skew_tick=1 console=ttyS1,115200 console=tty0 default_hugepagesz=1G hugepagesz=1G hugepages=40 hugepagesz=2M hugepages=0 ignition.platform.id=openstack amd_iommu=on iommu=pt irqaffinity=0-7 isolcpus=domain,nohz,managed_irq,8-127 nohz_full=8-127 rcu_nocbs=8-127 mce=off nohz=on net.ifnames=0 nowatchdog nmi_watchdog=0 nosoftlockup quiet rcu_nocb_poll rcupdate.rcu_cpu_stall_suppress=1 rcupdate.rcu_expedited=1 rcupdate.rcu_normal_after_boot=1 rcupdate.rcu_task_stall_timeout=0 rcutree.kthread_prio=99 security=selinux selinux=1
----

[#cpu-tuned-configuration]
=== CPU pinning via Tuned and kernel args

`tuned` is a system tuning tool that monitors system conditions to optimize performance using various predefined profiles. A key feature is its ability to isolate CPU cores for specific workloads, like real-time applications. This prevents the OS from utilizing these cores and potentially increasing latency.

> To enable and configure this feature, the first thing is to create a profile for the CPU cores we want to isolate. In this example, among 64 cores, we dedicate 60 cores (`1-30,33-62`) for the application and remaining 4 cores are used for housekeeping. Note that the design of isolated CPUs heavily depends on the applications.

Reviewer: ```suggestion
To enable and configure this feature, the first thing is to create a profile for the CPU cores we want to isolate. In this example, among 64 cores, we dedicate 60 cores (`1-30,33-62`) for the application and remaining 4 cores are used for housekeeping. Note that the design of isolated CPUs heavily depends on the real-time applications.
```

$ reboot
----

> To validate that the parameters are applied after the reboot, use the following command to check the kernel command line:

Reviewer: passive way is a doc recommendation IIRC
```suggestion
To validate that the parameters are applied after the reboot, can be used the following command to check the kernel command line:
```

Author: I dropped this change and reverted back to original

| pt
| This option enables the use of vfio for the DPDK interfaces.

> | intel_iommu or amd_iommu

Reviewer: we should test and_iommu before adding it

Author: This is AMD server and I believe it works :-)

edge7725-10:~ # dmesg | grep -Ei "DMAR | IOMMU"
[    0.026590] [     T0] Kernel command line: BOOT_IMAGE=/boot/vmlinuz-6.4.0-150700.51-default root=UUID=575291cf-74e8-42cf-8f2c-408a20dc00b8 skew_tick=1 console=ttyS1,115200 console=tty0 default_hugepagesz=1G hugepagesz=1G hugepages=40 hugepagesz=2M hugepages=0 ignition.platform.id=openstack **amd_iommu=on iommu=pt** irqaffinity=0-7 isolcpus=domain,nohz,managed_irq,8-127 nohz_full=8-127 rcu_nocbs=8-127 mce=off nohz=on net.ifnames=0 nowatchdog nmi_watchdog=0 nosoftlockup quiet rcu_nocb_poll rcupdate.rcu_cpu_stall_suppress=1 rcupdate.rcu_expedited=1 rcupdate.rcu_normal_after_boot=1 rcupdate.rcu_task_stall_timeout=0 rcutree.kthread_prio=99 security=selinux selinux=1
[    2.806069] [     T1] iommu: Default domain type: Passthrough (set via kernel command line)
[    2.824903] [     T1] pci 0000:c4:00.2: AMD-Vi: IOMMU performance counters supported
[    2.824987] [     T1] pci 0000:7e:01.0: **Adding to iommu group 0**
[    2.825023] [     T1] pci 0000:7e:02.0: Adding to iommu group 1
[    2.825098] [     T1] pci 0000:c4:00.3: Adding to iommu group 2
[    2.825138] [     T1] pci 0000:c4:01.0: Adding to iommu group 3
[    2.825159] [     T1] pci 0000:c4:01.1: Adding to iommu group 4
